### PR TITLE
LPD-40375 Allow Basic CRUD operations of a Display Page Templates Folder - delete, get and post

### PR DESCRIPTION
**Pull request by BarbaraCabrera**
## Motivation

[Link to ticket](https://liferay.atlassian.net/browse/LPD-40375)

### Changes

Add the basic operations (delete, get and post) for the Display Page Templates Folders.

## How to test it

1. Create a Display Page Template Folder
2. Go to [http://localhost:8080/o/api?endpoint=http://localhost:8080/o/headless-admin-site/v1.0/openapi.json](http://localhost:8080/o/api?endpoint=http://localhost:8080/o/headless-admin-site/v1.0/openapi.json)
3. Test the Get using the `siteExternalReferenceCode`and the `displayPageTemplateFolderExternalReferenceCode`. Copy the response to use it in the Post operation.
4. Test the Delete using the same information, the `siteExternalReferenceCode`and the `displayPageTemplateFolderExternalReferenceCode`
5. Test the Post with the information obtained in the step 3.

### CODE REVIEW

**Comment by liferay-page-management** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by BarbaraCabrera** [comment]
ci:test:sf

**Comment by BarbaraCabrera** [comment]
ci:test:relevant

File: modules/apps/headless/headless-admin-site/headless-admin-site-test/src/testIntegration/java/com/liferay/headless/admin/site/resource/v1_0/test/DisplayPageTemplateFolderResourceTest.java [hunk 1]
```diff
@@ -6,15 +6,411 @@
...
+		throws Exception {
+
+		super.
+			testPutSiteSiteExternalReferenceCodeDisplayPageTemplateFolderPermissionsPage();
+	}
+
+	@Override
+	protected String[] getAdditionalAssertFieldNames() {
+		return new String[] {"description", "name"};
+	}
```
Comment (ruben-pulido) [inline modules/apps/headless/headless-admin-site/headless-admin-site-test/src/testIntegration/java/com/liferay/headless/admin/site/resource/v1_0/test/DisplayPageTemplateFolderResourceTest.java:266]: We should keep these in sync with the fields that we are setting in the converter and specifying when creating/updatinga new entity. so they are also tested. For example, if we are setting externalReferenceCode, we should include it here, so it is verified by the tests.

**Review by ruben-pulido — CHANGES_REQUESTED** [review]

**Comment by BarbaraCabrera** [comment]
Hi, @ruben-pulido ! 😄 

After taking into consideration your comments I tried to use the external reference code instead of the id, but the main problem has come with the differences between the Display Page Template Folder (DPTF) and the Layout Page Template Collection (LPTC) entities. I can add the parent ERC to the DPTF, but the LPTC only have the parent Id, so every time I need to use the service I have to use the parent ERC to get the parent and then its id.

The thing is that when we can't find a collection with that parent external reference code, 

>`_layoutPageTemplateCollectionService.fetchLayoutPageTemplateCollection(parentDisplayPageTemplateFolderExternalReferenceCode, group.getGroupId()) == null`

I need to use the `PARENT_LAYOUT_PAGE_TEMPLATE_COLLECTION_ID_DEFAULT` but that "default collection" doesn't have an ERC because it's not a collection, it's just the parent id that we use to know that it doesn't really have a parent and that it's at the main level. So it means that I have to add null for the parent ERC and if we look at the validations that are generated:

image
 
That's not possible.

So the best solutions I can think of it's allow a parent ERC null (in the validations) or add an ERC for the default parent or leave the id.

I hope I explained it well, let me know what you think and thank you in advance! 🤓

**Comment by ruben-pulido** [comment]
https://github.com/liferay-page-management/liferay-portal/pull/16509

**Pull request closed without merging**